doc/source/ray-core/ray-generator.rst

```diff
@@ -0,0 +1,210 @@
+Since the ``StreamingObjectRefGenerator`` is just a Python generator, you can also simply use a for loop to
+iterate object references. 
+
+If you want to avoid blocking a thread, you can either use asyncio or :ref:`ray.wait API <generators-wait>`.
+
+.. literalinclude:: doc_code/streaming_generator.py
+    :language: python
+    :start-after: __streaming_generator_execute_start__
+    :end-before: __streaming_generator_execute_end__
+
+.. note::
+
+    For a normal Python generator, a generator function is paused and resumed when ``next`` function is
+    called on a generator. In Ray, a generator task is eagerly executed and sends the result back to the caller,
+    and it is never paused.
+
+Error Handling
+--------------
+
+If there's any failure from a generator task (by an application exception or system error such as unexpected node failure),
+the ``next(gen)`` will return an object reference that contains an exception. When you call ``ray.get``,
+the exception is raised. 
+
+.. literalinclude:: doc_code/streaming_generator.py
+    :language: python
+    :start-after: __streaming_generator_exception_start__
+    :end-before: __streaming_generator_exception_end__
+
+As you see from the example, If the task is failed by an application, the object reference with an exception
+is returned with at a correct order (for example, if the exception is raised after the second yield, the third 
+``next(gen)`` will return an object reference with an exception all the time). If the task is failed
+by a system error (e.g., node failure or worker process failure), the object reference that contains the system level exception
+can be returned from ``next(gen)`` any time without an ordering guarantee. 
+It means when you have N yields, the generator can create from 1 to N + 1 object references
+(N output + ref with a system-level exception) when there are failures.
+
+Generator from Actor Task
+-------------------------
+Ray generator is compatible with **all execution model of actors**. I.e., it seamlessly works with
+regular actor, :ref:`async actors <async-actors>`, and :ref:`threaded actors <threaded-actors>`.
+
+.. literalinclude:: doc_code/streaming_generator.py
+    :language: python
+    :start-after: __streaming_generator_actor_model_start__
+    :end-before: __streaming_generator_actor_model_end__
+
+Using Ray Generator with Asyncio
+--------------------------------
+The returned ``StreamingObjectRefGenerator`` is also compatible with asyncio, which means you can
+use ``__anext__`` or ``async for`` loop.
+
+.. literalinclude:: doc_code/streaming_generator.py
+    :language: python
+    :start-after: __streaming_generator_asyncio_start__
+    :end-before: __streaming_generator_asyncio_end__
+
+Garbage Collection of Object Referneces
+---------------------------------------
+The returned ref from ``next(generator)`` is just a regular Ray object reference, and it is distribute ref counted in the same way.
```

Comment:
This sentence seems to be grammatically incorrect. I am having trouble offering a suggestion without more context.